src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1166,6 +1171,34 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Diag
             return expression;
         }
 
+        /// <summary>
+        /// Is this is an _ identifier in a context where discards are allowed?
+        /// </summary>
+        private static bool FallBackOnDiscard(SimpleNameSyntax node, DiagnosticBag diagnostics)
+        {
+            if (!node.IsKind(SyntaxKind.IdentifierName) || node.Identifier.ContextualKind() != SyntaxKind.UnderscoreToken)
+            {
+                return false;
+            }
+
+            CSharpSyntaxNode containingDeconstruction = node.GetContainingDeconstruction();
+            return containingDeconstruction != null || IsOutVarDiscardIdentifier(node);
+        }
+
+        private static bool IsOutVarDiscardIdentifier(SimpleNameSyntax node)
+        {
+            Debug.Assert(node.Identifier.ContextualKind() == SyntaxKind.UnderscoreToken);
+
+            CSharpSyntaxNode parent = node.Parent;
+            return parent?.Kind() == SyntaxKind.Argument &&
+                ((ArgumentSyntax)parent).RefOrOutKeyword.Kind() == SyntaxKind.OutKeyword &&
+                (parent = parent.Parent)?.Kind() == SyntaxKind.ArgumentList &&
+                ((parent = parent.Parent)?.Kind() == SyntaxKind.InvocationExpression ||
```

Comment:
Once you've checked for the argument syntax's out keyword, I think you don't need any of the subsequent tests. The out keyword will only occur in an argument list (or there will be some other error).